frame/staking/src/tests.rs

```diff
@@ -3693,6 +3693,14 @@ fn do_not_die_when_active_is_ed() {
 		})
 }
 
+#[test]
+fn on_finalize_weight_is_nonzero() {
+	ExtBuilder::default().build_and_execute(|| {
+		let on_finalize_weight = <Test as frame_system::Config>::DbWeight::get().reads(1);
```

Comment:
```suggestion
		let on_finalize_weight = <Test as frame_system::Config>::DbWeight::get().reads_writes(1, 1);
```

Reply:
The code says you shouldnt count the write

Reply:
a read followed by a write is counted as just one read? 

We're also missing `T::UnixTime::new()`, which is one read in the timestamp module, but we should not assume that and return some weight.

Reply:
No, look at the code in `on_finalize` it says something like "because we only do this once per session, we discount the write operation from the weight.

Someone wrote that, and you seem to be ignoring it

Reply:
Right 🤦